src/modules/auth/controllers/registerController.ts

```diff
@@ -0,0 +1,36 @@
+import asyncHandler from "express-async-handler";
+import { Request, Response } from "express";
+import { User } from "../models/User";
+import { validateRegisterInput } from "../models/User";
+import { HttpStatus } from "../../common/enums/StatusCodes";
+
+export const registerUser = asyncHandler(
+  async (req: Request, res: Response): Promise<void> => {
+    const { firstname, lastname, state, email, password } = req.body;
+    const { error } = validateRegisterInput(req.body);
+    if (error) {
+      res.status(HttpStatus.BadRequest).json({
+        status: "Bad request",
+        message: "Registration unsuccessful",
```

Comment:
validation message not specific to any input error.
 Provide validation for all fields. When there’s a validation error, return status code 422 with payload:
 {
   "errors": [
     {
       "field": "string",
       "message": "string"
     },
   ]
 }